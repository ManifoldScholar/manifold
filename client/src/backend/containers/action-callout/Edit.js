import React, { Component } from "react";
import PropTypes from "prop-types";
import connectAndFetch from "utils/connectAndFetch";
import { actionCalloutsAPI, requests } from "api";
import { select } from "utils/entityUtils";
import Form from "./Form";
import { entityStoreActions } from "actions";

const { request } = entityStoreActions;

export class CallToActionEdit extends Component {
  static displayName = "CallToAction.Edit";

  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    match: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    calloutable: PropTypes.object.isRequired,
    closeRoute: PropTypes.string.isRequired,
    refreshActionCallouts: PropTypes.func,
    actionCallout: PropTypes.object
  };

  static mapStateToProps = (state, ownPropsIgnored) => {
    return {
      actionCallout: select(requests.beActionCallout, state.entityStore)
    };
  };

  componentDidMount() {
    this.fetchActionCallout(this.props.match.params.id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      this.fetchActionCallout(this.props.match.params.id);
    }
  }

  get calloutable() {
    return this.props.calloutable;
  }

  get actionCallout() {
    return this.props.actionCallout;
  }

  fetchActionCallout(id) {
    const call = actionCalloutsAPI.show(id);
    const actionCalloutRequest = request(call, requests.beActionCallout);
    this.props.dispatch(actionCalloutRequest);
  }

  render() {
    if (!this.actionCallout) return null;
    return (
      <Form
        refreshActionCallouts={this.props.refreshActionCallouts}
        actionCallout={this.actionCallout}
        closeRoute={this.props.closeRoute}
        calloutable={this.calloutable}
      />
    );
  }
}

export default connectAndFetch(CallToActionEdit);
