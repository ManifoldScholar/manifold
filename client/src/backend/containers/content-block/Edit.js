import React, { Component } from "react";
import PropTypes from "prop-types";
import connectAndFetch from "utils/connectAndFetch";
import { contentBlocksAPI, requests } from "api";
import { select } from "utils/entityUtils";
import Form from "backend/containers/project/content/Form";
import { entityStoreActions } from "actions";

const { request, flush } = entityStoreActions;

export class ContentBlockEditContainer extends Component {
  static displayName = "ContentBlock.Edit";

  static mapStateToProps = (state, ownPropsIgnored) => {
    return {
      contentBlock: select(requests.beContentBlock, state.entityStore)
    };
  };

  static propTypes = {
    match: PropTypes.object,
    history: PropTypes.object,
    dispatch: PropTypes.func.isRequired,
    project: PropTypes.object,
    contentBlock: PropTypes.object
  };

  componentDidMount() {
    this.fetchContentBlock(this.props.match.params.id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      this.fetchContentBlock(this.props.match.params.id);
    }
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.beContentBlock));
  }

  get contentBlock() {
    return this.props.contentBlock;
  }

  get project() {
    return this.props.project;
  }

  fetchContentBlock(id) {
    const call = contentBlocksAPI.show(id);
    const contentBlockRequest = request(call, requests.beContentBlock);
    this.props.dispatch(contentBlockRequest);
  }

  render() {
    if (!this.contentBlock) return null;

    return (
      <section>
        <header className="drawer-header">
          <h2 className="heading-quaternary">Edit Content Block</h2>
        </header>
        <Form contentBlock={this.contentBlock} project={this.project} />
      </section>
    );
  }
}

export default connectAndFetch(ContentBlockEditContainer);
