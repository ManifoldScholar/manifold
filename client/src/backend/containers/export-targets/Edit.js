import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import connectAndFetch from "utils/connectAndFetch";
import withConfirmation from "hoc/withConfirmation";
import { entityStoreActions } from "actions";
import { select } from "utils/entityUtils";
import { exportTargetsAPI, requests } from "api";
import lh from "helpers/linkHandler";
import Layout from "backend/components/layout";
import Form from "./Form";

const { request, flush } = entityStoreActions;

export class ExportTargetsEditContainer extends PureComponent {
  static mapStateToProps = state => {
    return {
      exportTarget: select(requests.beExportTarget, state.entityStore)
    };
  };

  static displayName = "ExportTargets.Edit";

  static propTypes = {
    exportTarget: PropTypes.object,
    match: PropTypes.object.isRequired,
    confirm: PropTypes.func.isRequired,
    dispatch: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    t: PropTypes.func
  };

  componentDidMount() {
    this.fetchExportTarget(this.exportTargetID);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.exportTargetID) {
      this.fetchExportTarget(this.exportTargetID);
    }
  }

  componentWillUnmount() {
    this.props.dispatch(flush([requests.beExportTarget]));
  }

  get exportTarget() {
    return this.props.exportTarget;
  }

  get exportTargetID() {
    return this.props.match.params.id;
  }

  destroyExportTarget = () => {
    const { dispatch, history } = this.props;
    const call = exportTargetsAPI.destroy(this.exportTargetID);
    const options = { removes: this.exportTarget };
    const exportTargetRequest = request(
      call,
      requests.beExportTargetDestroy,
      options
    );

    dispatch(exportTargetRequest).promise.then(() => {
      history.push(lh.link("backendSettingsExportTargets"));
    });
  };

  handleExportTargetDestroy = () => {
    const heading = this.props.t("modals.delete_export_target");
    const message = this.props.t("modals.confirm_body");

    this.props.confirm(heading, message, this.destroyExportTarget);
  };

  fetchExportTarget(id) {
    const call = exportTargetsAPI.show(id);
    const exportTargetRequest = request(call, requests.beExportTarget);

    this.props.dispatch(exportTargetRequest);
  }

  render() {
    if (!this.exportTarget) return null;
    const attr = this.exportTarget.attributes;

    return (
      <section>
        <Layout.DrawerHeader
          title={attr.name}
          buttons={[
            {
              onClick: this.handleExportTargetDestroy,
              icon: "delete32",
              label: this.props.t("actions.delete"),
              className: "utility-button__icon--notice"
            }
          ]}
        />
        <Form model={this.exportTarget} />
      </section>
    );
  }
}

export default withTranslation()(
  withConfirmation(connectAndFetch(ExportTargetsEditContainer))
);
