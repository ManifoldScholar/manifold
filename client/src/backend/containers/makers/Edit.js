import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import connectAndFetch from "utils/connectAndFetch";
import withConfirmation from "hoc/withConfirmation";
import { entityStoreActions } from "actions";
import { select } from "utils/entityUtils";
import { makersAPI, requests } from "api";
import get from "lodash/get";
import Form from "./Form";
import lh from "helpers/linkHandler";
import Layout from "backend/components/layout";

const { request, flush } = entityStoreActions;

export class MakersEditContainer extends PureComponent {
  static mapStateToProps = (state, ownPropsIgnored) => {
    return {
      maker: select(requests.beMaker, state.entityStore),
      updateMakers: get(state.entityStore.responses, requests.beMakerUpdate)
    };
  };

  static displayName = "Makers.Edit";

  static propTypes = {
    maker: PropTypes.object,
    confirm: PropTypes.func.isRequired,
    match: PropTypes.object,
    history: PropTypes.object,
    afterDestroy: PropTypes.func,
    dispatch: PropTypes.func,
    t: PropTypes.func,
    refetch: PropTypes.func.isRequired
  };

  static defaultProps = {
    confirm: (heading, message, callback) => callback()
  };

  componentDidMount() {
    this.fetchMaker(this.props.match.params.id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      this.fetchMaker(this.props.match.params.id);
    }
  }

  componentWillUnmount() {
    this.props.dispatch(flush([requests.beMakerUpdate]));
  }

  fetchMaker(id) {
    const call = makersAPI.show(id);
    const makerRequest = request(call, requests.beMaker);
    this.props.dispatch(makerRequest);
  }

  handleMakerDestroy = () => {
    const heading = this.props.t("modals.delete_maker");
    const message = this.props.t("modals.confirm_body");
    this.props.confirm(heading, message, this.destroyMaker);
  };

  destroyMaker = () => {
    const maker = this.props.maker;
    const call = makersAPI.destroy(maker.id);
    const makerRequest = request(call, requests.beMakerDestroy);
    this.props.dispatch(makerRequest).promise.then(() => {
      this.doAfterDestroy(this.props);
    });
  };

  doAfterDestroy(props) {
    this.props.refetch();
    if (props.afterDestroy) return props.afterDestroy();
    return props.history.push(lh.link("backendRecordsMakers"));
  }

  render() {
    if (!this.props.maker) return null;
    const attr = this.props.maker.attributes;
    const maker = this.props.maker;

    return (
      <div>
        <Layout.DrawerHeader
          title={attr.fullName}
          buttons={[
            {
              onClick: this.handleMakerDestroy,
              icon: "delete32",
              label: this.props.t("actions.delete"),
              entity: maker,
              ability: "delete",
              className: "utility-button__icon--notice"
            }
          ]}
        />
        <Form maker={maker} />
      </div>
    );
  }
}

export default withTranslation()(
  withConfirmation(connectAndFetch(MakersEditContainer))
);
