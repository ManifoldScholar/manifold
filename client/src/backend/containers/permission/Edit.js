import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import Form from "./Form";
import { permissionsAPI, requests } from "api";
import { entityStoreActions } from "actions";
import connectAndFetch from "utils/connectAndFetch";
import { select } from "utils/entityUtils";
import Layout from "backend/components/layout";
import withConfirmation from "hoc/withConfirmation";

const { request, flush } = entityStoreActions;

export class PermissionEdit extends PureComponent {
  static mapStateToProps = state => {
    return {
      permission: select(requests.bePermission, state.entityStore)
    };
  };

  static displayName = "Permission.Edit";

  static propTypes = {
    entity: PropTypes.object,
    permission: PropTypes.object,
    match: PropTypes.object,
    dispatch: PropTypes.func,
    closeUrl: PropTypes.string.isRequired,
    history: PropTypes.object,
    confirm: PropTypes.func.isRequired,
    t: PropTypes.func
  };

  static defaultProps = {
    confirm: (heading, message, callback) => callback()
  };

  componentDidMount() {
    this.fetchPermission(this.props.match.params.id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      this.fetchPermission(this.props.match.params.id);
    }
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.bePermission));
  }

  fetchPermission(id) {
    const entity = this.props.entity;
    const action = request(
      permissionsAPI.show(entity, id),
      requests.bePermission
    );
    this.props.dispatch(action);
  }

  handleRemoveAll = () => {
    const t = this.props.t;
    const heading = t("modals.delete_permissions");
    const message = t("modals.confirm_body");
    this.props.confirm(heading, message, this.removeAllPermissions);
  };

  removeAllPermissions = () => {
    event.preventDefault();
    const entity = this.props.entity;
    const permission = this.props.permission;
    const options = { removes: permission };
    const action = request(
      permissionsAPI.destroy(entity, permission.id),
      requests.bePermissionDestroy,
      options
    );
    this.props.dispatch(action).promise.then(() => {
      this.props.history.push(this.props.closeUrl);
    });
  };

  render() {
    const { permission, t } = this.props;
    if (!permission) return null;
    return (
      <section>
        <Layout.DrawerHeader
          title={t("projects.permissions.edit_header")}
          buttons={[
            {
              onClick: this.handleRemoveAll,
              icon: "delete32",
              label: t("actions.delete"),
              className: "utility-button__icon--notice"
            }
          ]}
        />
        <Form
          entity={this.props.entity}
          permission={permission}
          history={this.props.history}
          showUserInput
        />
      </section>
    );
  }
}

export default withTranslation()(
  withConfirmation(connectAndFetch(PermissionEdit))
);
