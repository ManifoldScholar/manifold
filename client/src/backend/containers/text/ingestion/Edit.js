import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import connectAndFetch from "utils/connectAndFetch";
import Ingestion from "backend/components/ingestion";
import { entityStoreActions } from "actions";
import { ingestionsAPI, requests } from "api";
import { select, isLoaded } from "utils/entityUtils";
import lh from "helpers/linkHandler";

const { request } = entityStoreActions;

export class IngestionEditContainer extends PureComponent {
  static fetchData = (getState, dispatch, location, match) => {
    if (isLoaded(requests.beIngestionShow, getState())) return;
    const call = ingestionsAPI.show(match.params.ingestionId);
    const ingestion = request(call, requests.beIngestionShow);
    return dispatch(ingestion);
  };

  static mapStateToProps = state => {
    return {
      ingestion: select(requests.beIngestionShow, state.entityStore)
    };
  };

  static displayName = "Text.Ingestion.Edit";

  static propTypes = {
    text: PropTypes.object.isRequired,
    history: PropTypes.object,
    ingestion: PropTypes.object,
    match: PropTypes.object,
    location: PropTypes.object
  };

  get textId() {
    return this.props.text.id;
  }

  handleSuccess = () => {
    this.redirectToIngestion(this.props.ingestion.id);
  };

  redirectToIngestion(ingestionId) {
    const path = lh.link(
      "backendTextIngestionIngest",
      this.textId,
      ingestionId
    );
    this.props.history.push(path, { back: this.props.match.url });
  }

  render() {
    return (
      <div>
        {this.props.ingestion ? (
          <Ingestion.Form.Wrapper
            ingestion={this.props.ingestion}
            location={this.props.location}
            history={this.props.history}
            name={requests.beIngestionCreate}
            project={this.props.text.relationships.project}
            onSuccess={this.handleSuccess}
          />
        ) : null}
      </div>
    );
  }
}

export default connectAndFetch(IngestionEditContainer);
