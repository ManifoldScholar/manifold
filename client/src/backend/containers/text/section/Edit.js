import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import Navigation from "backend/components/navigation";
import AddEditSectionForm from "backend/components/authoring/AddEditSectionForm";
import { useParams } from "react-router-dom";
import { useFetch } from "hooks";
import { sectionsAPI } from "api";

export default function EditSectionContainer({ text }) {
  const { t } = useTranslation();
  const { sectionId } = useParams();

  const { data: section } = useFetch({
    request: [sectionsAPI.show, sectionId]
  });

  return (
    <section>
      <Navigation.DrawerHeader
        title={t("backend_entities.texts.edit_section")}
      />
      <AddEditSectionForm text={text} section={section} />
    </section>
  );
}

EditSectionContainer.displayName = "Text.Sections.Edit";

EditSectionContainer.propTypes = {
  text: PropTypes.object.isRequired
};
