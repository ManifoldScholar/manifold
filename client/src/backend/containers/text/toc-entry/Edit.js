import React from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import Navigation from "backend/components/navigation";
import EditTOCEntryForm from "backend/components/authoring/AddEditTOCEntryForm";

export default function EditTOCEntryContainer(props) {
  const { t } = useTranslation();

  return (
    <section>
      <Navigation.DrawerHeader
        title={t("backend.forms.text_toc.edit_header")}
      />
      <EditTOCEntryForm {...props} />
    </section>
  );
}

EditTOCEntryContainer.displayName = "Text.TOC.Edit";

EditTOCEntryContainer.propTypes = {
  entryId: PropTypes.string.isRequired,
  textId: PropTypes.string.isRequired,
  sections: PropTypes.array.isRequired,
  toc: PropTypes.array,
  setTree: PropTypes.func.isRequired
};
