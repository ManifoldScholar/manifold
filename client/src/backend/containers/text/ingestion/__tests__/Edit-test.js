import { IngestionEditContainer } from "../Edit";

describe("backend/containers/text/ingestion/Edit", () => {
  def("text", () => factory("text"));

  def("root", () => (
    <IngestionEditContainer
      text={$text}
      location={{}}
      history={fixtures.history()}
    />
  ));

  it("matches the snapshot when rendered", () => {
    expect(mount($withApp($root)).html()).toMatchSnapshot();
  });
});
