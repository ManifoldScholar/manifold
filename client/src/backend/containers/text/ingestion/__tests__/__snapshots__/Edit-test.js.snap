// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backend/containers/text/ingestion/Edit matches the snapshot when rendered 1`] = `"<div></div>"`;
